ui/app/pages/confirmation/confirmation.js

```diff
@@ -0,0 +1,243 @@
+import SiteIcon from '../../components/ui/site-icon';
+import { DEFAULT_ROUTE } from '../../helpers/constants/routes';
+import { stripHttpSchemes } from '../../helpers/utils/util';
+import { useI18nContext } from '../../hooks/useI18nContext';
+import { useOriginMetadata } from '../../hooks/useOriginMetadata';
+import { getUnapprovedConfirmations } from '../../selectors';
+import NetworkDisplay from '../../components/app/network-display/network-display';
+import { COLORS, SIZES } from '../../helpers/constants/design-system';
+import Callout from '../../components/ui/callout';
+import { getTemplateValues, getTemplateAlerts } from './templates';
+
+/**
+ * a very simple reducer using produce from Immer to keep state manipulation
+ * immutable and painless. This state is not stored in redux state because it
+ * should persist only for the lifespan of the current session, and will only
+ * be used on this page. Dismissing alerts for confirmations should persist
+ * while the user pages back and forth between confirmations. However, if the
+ * user closes the confirmation window and later reopens the extension they
+ * should be displayed the alerts again.
+ */
+const alertStateReducer = produce((state, action) => {
+  switch (action.type) {
+    case 'dismiss':
+      if (state?.[action.confirmationId]?.[action.alertId]) {
+        state[action.confirmationId][action.alertId].dismissed = true;
+      }
+      break;
+    case 'set':
+      if (!state[action.confirmationId]) {
+        state[action.confirmationId] = {};
+      }
+      action.alerts.forEach((alert) => {
+        state[action.confirmationId][alert.id] = {
+          ...alert,
+          dismissed: false,
+        };
+      });
+      break;
+    default:
+      throw new Error(
+        'You must provide a type when dispatching an action for alertState',
+      );
+  }
+});
+
+/**
+ * Encapsulates the state and effects needed to manage alert state for the
+ * confirmation page in a custom hook. This hook is not likely to be used
+ * outside of this file, but it helps to reduce complexity of the primary
+ * component.
+ * @param {Object} pendingConfirmation - a pending confirmation waiting for
+ *  user approval
+ * @returns {[alertState: Object, dismissAlert: Function]} - tuple with
+ *  the current alert state and function to dismiss an alert by id
+ */
+function useAlertState(pendingConfirmation) {
+  const [alertState, dispatch] = useReducer(alertStateReducer, {});
+
+  /**
+   * Computation of the current alert state happens every time the current
+   * pendingConfirmation changes. The async function getTemplateAlerts is
+   * responsible for returning alert state. Setting state on unmounted
+   * components is an anti-pattern, so we use a signal variable to keep track
+   * of the current state of the component. Returning a function that sets
+   * signal to false when the component is unmounted.
+   */
+  useEffect(() => {
+    let signal = true;
```

Comment:
Nit: Why "signal"? :thinking: I've seen something like this done before using an `isMounted` state. Maybe `isMounted` would be a better name for this?

Reply:
isMounted works, I was mimicking a useCancelablePromise hook I've written in the past when I wrote this. I'll reword it

Reply:
resolved in latest commit.